core/src/main/java/jenkins/telemetry/impl/SecuritySystemProperties.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.telemetry.impl;
+
+import hudson.Extension;
+import jenkins.model.Jenkins;
+import jenkins.telemetry.Telemetry;
+import jenkins.util.SystemProperties;
+import net.sf.json.JSONObject;
+
+import javax.annotation.Nonnull;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.Date;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.TreeMap;
+
+/**
+ * Telemetry implementation gathering information about system properties.
+ */
+@Extension
+public class SecuritySystemProperties extends Telemetry {
+    @Nonnull
+    @Override
+    public String getId() {
+        return "security-system-properties";
+    }
+
+    @Nonnull
+    @Override
+    public LocalDate getStart() {
+        return LocalDate.of(2018, 9, 1);
+    }
+
+    @Nonnull
+    @Override
+    public LocalDate getEnd() {
+        return LocalDate.of(2018, 12, 1);
+    }
+
+    @Nonnull
+    @Override
+    public String getDisplayName() {
+        return "Use of Security-related Java system properties";
+    }
+
+    @Nonnull
+    @Override
+    public JSONObject createContent() {
+        Map<String, String> security = new TreeMap<>();
+        putBoolean(security, "hudson.ConsoleNote.INSECURE", false);
+        putBoolean(security, "hudson.model.ParametersAction.keepUndefinedParameters", false);
+        putBoolean(security, "hudson.model.User.allowNonExistentUserToLogin", false);
+        putBoolean(security, "hudson.model.User.allowUserCreationViaUrl", false);
+        putBoolean(security, "hudson.model.User.SECURITY_243_FULL_DEFENSE", true);
+
+        putStringInfo(security, "hudson.model.ParametersAction.safeParameters");
```

Comment:
Not sure it is really anonymous. It may contain classes of private-source plugins, for example

Reply:
We only collect whether it is defined, and, if it is, the length. Same with CSP.